Added SSH and Kubernetes resource (#838)
Twingate/terraform-provider-twingate: docs/resources/ssh_resource.md

- `tags` (Map of String) A map of key-value pair tags to set on this resource.
- `username` (String) The username to use when connecting to the SSH Resource.

### Read-Only

- `id` (String) Autogenerated ID of the SSH Resource.

<a id="nestedblock--access_group"></a>
### Nested Schema for `access_group`

Optional:

- `access_policy` (Block Set) Restrict access according to JIT access policy (see [below for nested schema](#nestedblock--access_group--access_policy))
- `group_id` (String) Group ID that will have permission to access the Resource.
- `security_policy_id` (String) The ID of a `twingate_security_policy` to use as the access policy for the group IDs in the access block. Default is 'Null' which points to `Default Policy` on Admin console.

<a id="nestedblock--access_group--access_policy"></a>
### Nested Schema for `access_group.access_policy`

Optional:

- `approval_mode` (String) This will set the approval model on the edge. The valid values are `AUTOMATIC` and `MANUAL`.
- `duration` (String) This will set the access duration on the edge. Duration must be between 1 hour and 365 days. The valid values are like `1h` and `2d`.

> - `mode` (String) This will set the access_policy mode on the edge. The valid values are `MANUAL`, `AUTO_LOCK` and `ACCESS_REQUEST`.

Reviewer: ```suggestion
- `mode` (String) This will set the access_policy mode for the policy. The valid values are `MANUAL`, `AUTO_LOCK` and `ACCESS_REQUEST`.
```

<a id="nestedblock--access_policy"></a>
### Nested Schema for `access_policy`

Optional:

> - `approval_mode` (String) This will set the approval model on the edge. The valid values are `AUTOMATIC` and `MANUAL`.

Reviewer: ```suggestion
- `approval_mode` (String) This will set the approval model for the policy. The valid values are `AUTOMATIC` and `MANUAL`.
```

---
# generated by https://github.com/hashicorp/terraform-plugin-docs
page_title: "twingate_ssh_resource Resource - terraform-provider-twingate"
subcategory: ""
description: |-

>   SSH Resources are Twingate resources accessed via an SSH Gateway.

Reviewer 2: `...via a Gateway.`

---

# twingate_ssh_resource (Resource)

> SSH Resources are Twingate resources accessed via an SSH Gateway.

Reviewer 2: Ditto

  gateway_id = twingate_gateway.main.id
  remote_network_id = twingate_remote_network.prod.id
  address    = "10.128.0.106"
  username   = "ubuntu-2"
}

resource "local_file" "config" {
  content  = templatefile("${path.module}/config-template.yaml", {
    ssh_resources = [twingate_ssh_resource.bastion, twingate_ssh_resource.bastion2]
    kubernetes_resources = [twingate_kubernetes_resource.prod_cluster]
  })
  filename = "${path.module}/generated/config.yaml"
}
```

### config-template.yaml
```yaml
twingate:
network: "my-network"
host: "twingate.com"

port: 8443
metricsPort: 9090

tls:
certificateFile: "/etc/gateway/tls.crt"
privateKeyFile: "/etc/gateway/tls.key"

kubernetes:
upstreams: %{ for item in kubernetes_resources }
- name: ${item.name}
address: ${item.address}
inCluster: ${item.in_cluster}%{ endfor }

ssh:
gateway:
username: "gateway"
key:
type: "ed25519"
hostCertificate:
ttl: "24h"
userCertificate:
ttl: "5m"

ca:
vault:
address: "<https://vault.example.com:8200>"
caBundleFile: "/etc/ssl/vault-ca.crt"
auth:
token: "<vault-token>"
mount: "ssh"
role: "gateway"

upstreams: %{ for item in ssh_resources }
- name: ${item.name}
address: ${item.address}

> user: ${item.username}%{ endfor }

Reviewer 2: nit: indentation is not correct
